pagespeed/envoy/envoy_fetch.cc

```diff
@@ -0,0 +1,143 @@
+
+#include <algorithm>
+#include <list>
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+
+#include "net/instaweb/http/public/async_fetch.h"
+#include "net/instaweb/http/public/inflating_fetch.h"
+#include "pagespeed/envoy/header_utils.h"
+#include "pagespeed/kernel/base/basictypes.h"
+#include "pagespeed/kernel/base/condvar.h"
+#include "pagespeed/kernel/base/message_handler.h"
+#include "pagespeed/kernel/base/pool.h"
+#include "pagespeed/kernel/base/pool_element.h"
+#include "pagespeed/kernel/base/statistics.h"
+#include "pagespeed/kernel/base/string_util.h"
+#include "pagespeed/kernel/base/thread_system.h"
+#include "pagespeed/kernel/base/timer.h"
+#include "pagespeed/kernel/base/writer.h"
+#include "pagespeed/kernel/http/request_headers.h"
+#include "pagespeed/kernel/http/response_headers.h"
+#include "pagespeed/kernel/http/response_headers_parser.h"
+#include "pagespeed_remote_data_fetcher.h"
+
+namespace net_instaweb {
+
+// Default keepalive 60s.
+const int64 keepalive_timeout_ms = 60000;
+static const char cluster_str[] = "cluster1";
+
+PagespeedDataFetcherCallback::PagespeedDataFetcherCallback(EnvoyFetch* fetch) { fetch_ = fetch; }
+
+void PagespeedDataFetcherCallback::onSuccess(Envoy::Http::MessagePtr& response) {
+  fetch_->setResponse(response->headers(), response->body());
+}
+
+void PagespeedDataFetcherCallback::onFailure(FailureReason reason) {
+  std::cout << "PagespeedDataFetcherCallback::onFailure\n";
+  std::cout.flush();
+}
+
+EnvoyFetch::EnvoyFetch(const GoogleString& url,
+                   AsyncFetch* async_fetch,
+                   MessageHandler* message_handler,
+                   EnvoyClusterManager& cluster_manager)
+    : str_url_(url),
+      fetcher_(NULL),
+      async_fetch_(async_fetch),
+      message_handler_(message_handler),
+      cluster_manager_(cluster_manager),
+      done_(false),
+      content_length_(-1),
+      content_length_known_(false) {
+}
+
+EnvoyFetch::~EnvoyFetch() {
+}
+
+void EnvoyFetch::FetchWithEnvoy() {
+
+  envoy::api::v2::core::HttpUri http_uri;
+  http_uri.set_uri(str_url_);
+
+  http_uri.set_cluster(cluster_str);
+  cb_ptr_ = std::make_unique<PagespeedDataFetcherCallback>(this);
+  std::unique_ptr<PagespeedRemoteDataFetcher> PagespeedRemoteDataFetcherPtr = 
+      std::make_unique<PagespeedRemoteDataFetcher>(cluster_manager_.getClusterManager(str_url_), http_uri, *cb_ptr_);
+
+  PagespeedRemoteDataFetcherPtr->fetch();
+  cluster_manager_.getDispatcher()->run(Envoy::Event::Dispatcher::RunType::Block);
+}
+
+// This function is called by EnvoyUrlAsyncFetcher::StartFetch.
+void EnvoyFetch::Start() {
+  std::function<void()> fetch_fun_ptr = std::bind(&EnvoyFetch::FetchWithEnvoy, this);
+  cluster_manager_.getDispatcher()->post(fetch_fun_ptr);
+  cluster_manager_.getDispatcher()->run(Envoy::Event::Dispatcher::RunType::NonBlock);
+}
+
+bool EnvoyFetch::Init() {
+  return true;
+}
+// This function should be called only once. The only argument is sucess or
```

Comment:
nit: Newline between `}` and `//...` here and below

Reply:
Removed comment along with the function, as the function is not used in fetcher. I added this when I took functionality from serf fetcher, but not using anymore.
Code pushed in https://github.com/apache/incubator-pagespeed-mod/pull/1969/commits/8a92117ec74a99b4a1dffff479a71ca1b5db7717